### PR TITLE
Sorting publications on frontend

### PR DESCRIPTION
**Pull request by vgoh1**
# Description 
Implement sorting of publications by year, author, publication title and category (journal/conference) title

# Type of change 
- [ ] Bug fix
- [x] New feature
- [ ] Refactoring
- [ ] Documentation 

# Problem
Client wants a way to sort publications 

# Solution description
There's a button that calls the sorting function to be done on the publications list, and also keeps track of what it's being sorted by. The default sort is by year, the query from the DB already comes sorted by year (newest to oldest). There isn't a way to sort by category title (journal/conference) for the list view because it seems redundant when you can just change the layout to do the split. Also if anyone has any suggestions for something better sounding than category title, suggest it pls. 

# Tests 
The sorting for the list view is self explanatory so I won't be including screenshots to those, but for the journal/conference split I will:
image

image

image

'Some conference' comes before 'Some journal'
image

# Relevant document/ link (if any) 
Any related documents could help reviewers to review this PR 
N/A

# Risk
Identity the risk level (low, medium, high) and indicate which component(s) will be affected if bugs exist.
Low, should only affect the front end view for publications

### CODE REVIEW

**Comment by rsun0013** [comment]
Looks good

**Review by rsun0013 — APPROVED** [review]

File: api/src/controllers/publications.js [hunk 1]
```diff
@@ -136,7 +136,19 @@ async function readAllPublicationsByTeam(req, res) {
         }
     }
 
-    const foundPublication = await Publication.find({ teamId: _id });
+    // const foundPublication = await Publication.find({ teamId: _id }).sort([["yearPublished", -1]]);
```
Comment (aksh0001) [inline api/src/controllers/publications.js:139]: Can we remove this comment?

File: client/src/components/publications/Publications.js [hunk 2]
```diff
@@ -40,6 +41,15 @@ const Publications = () => {
         }
     }
 
+    const toggleSortingOptions = () => {
+        switch(layout) {
+            case allLayouts.byCategory:
+                return <Dropdown.Item as="button" value="Category Title" onClick={e => {dispatch(sortPublications(teamPublications, e.target.value)); setSortingOption(e.target.value)}}>Category Title</Dropdown.Item>
```
Comment (aksh0001) [inline client/src/components/publications/Publications.js:47]: This line can be spanned over multiple lines as it its a bit too long.

File: client/src/components/publications/Publications.js [hunk 3]
```diff
@@ -75,6 +85,13 @@ const Publications = () => {
                                     }
                                 </Dropdown.Menu>
                             </Dropdown>
+                                
+                            <DropdownButton variant="light" id="dropdown-item-button" title={"Sort by: "+sortingOption} >
+                                <Dropdown.Item as="button" value="Year" onClick={e => {dispatch(sortPublications(teamPublications, e.target.value)); setSortingOption(e.target.value)}}>Year</Dropdown.Item>
+                                <Dropdown.Item as="button" value="Author" onClick={e => {dispatch(sortPublications(teamPublications, e.target.value)); setSortingOption(e.target.value)}}>Author</Dropdown.Item>
+                                <Dropdown.Item as="button" value="Title" onClick={e => {dispatch(sortPublications(teamPublications, e.target.value)); setSortingOption(e.target.value)}}>Title</Dropdown.Item>
```
Comment (aksh0001) [inline client/src/components/publications/Publications.js:92]: same as above.

**Pull request merged**